src/bin/pg_waldump/pg_waldump.c

```diff
@@ -628,18 +628,20 @@ XLogDumpDisplayRecord(XLogDumpConfig *config, XLogReaderState *record)
 	XLogRecGetLen(record, &rec_len, &fpi_len);
 
 	if(private->input_filename)
-		printf("rmgr: %-11s len (rec/tot): %6u/%6u, tx: %10u, offset: 0x%lX, prev %X/%08X, ",
+		printf("rmgr: %-11s len (rec/tot): %6u/%6u, tx: %10u, offset: 0x%lX, end: %X/%08X prev %X/%08X, ",
```

Comment:
This is implied by offset + len.

Why do we need this specifically?

Reply:
right. added this end so that we can grep end directly. It's not a must have like you said. 

Do you see any harm printing end?

Reply:
No strong opinion here. :-)  Either way works.